src/ClientModel/CaptureData.cpp

```diff
@@ -175,15 +175,21 @@ const std::string& CaptureData::GetFunctionNameByAddress(uint64_t absolute_addre
 // callstack.
 std::optional<uint64_t> CaptureData::FindFunctionAbsoluteAddressByAddress(
     uint64_t absolute_address) const {
-  const FunctionInfo* function = FindFunctionByAddress(absolute_address, false);
-  if (function != nullptr) {
-    return GetAbsoluteAddress(*function);
-  }
-  const LinuxAddressInfo* address_info = GetAddressInfo(absolute_address);
-  if (address_info != nullptr) {
-    return absolute_address - address_info->offset_in_function();
-  }
-  return std::nullopt;
+  const auto module_or_error = process_.FindModuleByAddress(absolute_address);
+  if (module_or_error.has_error()) return std::nullopt;
+  const std::string& module_path = module_or_error.value().file_path();
+  const std::string& module_build_id = module_or_error.value().build_id();
+  const uint64_t module_base_address = module_or_error.value().start();
+
+  const ModuleData* module =
+      module_manager_->GetModuleByPathAndBuildId(module_path, module_build_id);
+  if (module == nullptr) return std::nullopt;
+
+  const uint64_t offset = absolute_address - module_base_address;
+  const auto* function_info = module->FindFunctionByOffset(offset, false);
+  if (function_info == nullptr) return std::nullopt;
+
+  return module_base_address - module->load_bias() + function_info->address();
```

Comment:
I kind of accidentally dropped address-info related logic from here. Will need to revisit.

Reply:
done.